tools/pot/openvino/tools/pot/data_loaders/syntetic_image_loader.py

```diff
@@ -0,0 +1,293 @@
+                    break
+
+            self.xs.append(next_x)
+            self.ys.append(next_y)
+            prev_x = next_x
+            prev_y = next_y
+
+    @staticmethod
+    def process_nans(data):
+        nan_index = np.where(np.isnan(data))
+        extend = np.array(range(nan_index[0][0] - 100, nan_index[0][0]))
+        delete_row = np.append(extend, nan_index)
+        return delete_row
+
+    def rescale(self, image_x, image_y, pad_x, pad_y):
+        xs = np.array(self.xs)
+        ys = np.array(self.ys)
+        if np.any(np.isnan(xs)):
+            delete_row = self.process_nans(xs)
+            xs = np.delete(xs, delete_row, axis=0)
+            ys = np.delete(ys, delete_row, axis=0)
+
+        if np.any(np.isnan(ys)):
+            delete_row = self.process_nans(ys)
+            xs = np.delete(xs, delete_row, axis=0)
+            ys = np.delete(ys, delete_row, axis=0)
+
+        if np.min(xs) < 0.0:
+            xs -= np.min(xs)
+        if np.min(ys) < 0.0:
+            ys -= np.min(ys)
+        xmax, xmin = np.max(xs), np.min(xs)
+        ymax, ymin = np.max(ys), np.min(ys)
+        self.xs = np.uint16(xs / (xmax - xmin) * (image_x - 2 * pad_x) + pad_x)
+        self.ys = np.uint16(ys / (ymax - ymin) * (image_y - 2 * pad_y) + pad_y)
+
+    def draw(self, draw_type, image_x, image_y, pad_x=6, pad_y=6):
+        self.rescale(image_x, image_y, pad_x, pad_y)
+        image = np.empty((image_x, image_y, 3 if draw_type == 'color' else 1))
+
+        for i in range(len(self.xs)):
+            mask_pattern = '{:09b}'.format(np.random.randint(1, 512))
+            if draw_type == 'color':
+                patch = self.make_patch3_3(mask_pattern, [self.convert_color(i, 128)])
+            elif draw_type == 'point':
+                patch = np.array([127, 127, 127])
+            else:
+                patch = self.make_patch3_3(mask_pattern, [127, 127, 127])
+
+            # why + 1????
+            if draw_type == 'point':
+                image = np.array(image)
+                image[self.ys[i] + 1, self.xs[i] + 1, :] = 127, 127, 127
+                # image.paste(patch, (self.xs[i], self.ys[i]))
+            else:
+                H, W, C = patch.shape
+                x_start = self.xs[i] + 1
+                y_start = self.ys[i] + 1
+                image[x_start : x_start + H, y_start : y_start + W, :] = patch
+                # Image.fromarray(patch)
+                # image.paste(patch, (self.xs[i] + 1, self.ys[i] + 1))
+
+        image = np.array(image, dtype='uint8')
+        image = cv.cvtColor(image, cv.COLOR_RGB2BGR)
+        return image
+
+    def make_patch3_3(self, mask, patch_color):
+        patch_color = np.array(patch_color)
+        patch = np.zeros((3, 3, 3), np.uint8)
+        for i in range(3):
+            for j in range(3):
+                patch[i ,j, :] = patch_color * int(mask[i*3+j])
+        return patch
+
+
+class SynteticImageLoader(ImageLoader):
```

Comment:
Please also rename `syntetic` to `synthetic` everywhere (in the name of the file as well).